kitsune/inproduct/admin.py

```diff
@@ -1,13 +1,84 @@
+from django.conf import settings
 from django.contrib import admin
+from django.utils.html import format_html
+from django.utils.translation import gettext_lazy as _
 
 from kitsune.inproduct.models import Redirect
+from kitsune.wiki.models import Revision
+
+
+class KBLinkFilter(admin.SimpleListFilter):
+    title = _("KB article")
+    parameter_name = "kb_article"
+
+    def lookups(self, request, model_admin):
+        return (
+            ("with_kb", _("Has KB Article")),
+            ("without_kb", _("No KB Article")),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == "with_kb":
+            # Filter for objects with a KB link
+            return queryset.filter(id__in=[obj.id for obj in queryset if self.has_kb_link(obj)])
+        if self.value() == "without_kb":
+            # Filter for objects without a KB link
+            return queryset.exclude(id__in=[obj.id for obj in queryset if self.has_kb_link(obj)])
+
+    def has_kb_link(self, obj):
+        if "kb/" in obj.target:
+            try:
+                doc_slug = obj.target.split("/")[-1]
+                if obj.locale:
+                    locale = obj.locale
+                else:
+                    locale = settings.WIKI_DEFAULT_LANGUAGE
+                rev = Revision.objects.filter(
+                    document__slug=doc_slug, document__locale=locale
+                ).exists()
+                return rev
+            except Revision.DoesNotExist:
+                pass
+        return False
 
 
 class RedirectAdmin(admin.ModelAdmin):
-    list_display = ("product", "version", "platform", "locale", "topic", "target")
+    list_display = (
+        "product",
+        "version",
+        "platform",
+        "locale",
+        "topic",
+        "target",
+        "kb_link",
+    )
     list_display_links = ("topic", "target")
-    list_filter = ("product", "version", "platform", "locale")
+    list_filter = ("product", "version", "platform", "locale", KBLinkFilter)
     search_fields = ("topic", "target")
 
+    @admin.display(description="KB Article")
+    def kb_link(self, obj):
+        if "kb/" in obj.target:
+            if obj.locale:
+                rev = (
+                    Revision.objects.get(
+                        document__slug=obj.target.split("/")[-1], document__locale=obj.locale
+                    )
+                    or None
+                )
+            else:
+                rev = (
+                    Revision.objects.get(
+                        document__slug=obj.target.split("/")[-1],
+                        document__locale=settings.WIKI_DEFAULT_LANGUAGE,
+                    )
+                    or None
+                )
+            if rev:
+                target_url = rev.document.get_absolute_url()
+                return format_html('<a href="{}">{}</a>', target_url, target_url)
+        else:
+            return None
```

Comment:
There are two things I think that would make this better. First, I'd make the initial check tighter, more consistent with the actual redirect code, which strips out a leading "/" if there is one. Second,  I'd make this more DRY, and I think it'd be fine, as well as simpler and faster, to search `Document.objects` instead.  

Maybe something like:
```suggestion
        target = obj.target.lstrip("/")

        if not target.startswith("kb/"):
            return None

        slug = target.split("/")[-1]
        locale = obj.locale if obj.locale else settings.WIKI_DEFAULT_LANGUAGE

        try:
            doc = Document.objects.get(slug=slug, locale=locale)
        except Document.DoesNotExist:
            return None
        else:
            target_url = doc.get_absolute_url()

        return format_html('<a href="{}">{}</a>', target_url, target_url)
```